kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -436,6 +440,124 @@ private static Map<Long, StructField> fieldsById(StructType schema) {
     return columnIdToField;
   }
 
+  private static void validateNestedIds(
+      SchemaChanges schemaChanges,
+      Map<Long, StructField> currentFieldsById,
+      Map<Long, StructField> updatedFieldsById) {
+    for (Tuple2<StructField, StructField> updatedField : schemaChanges.updatedFields()) {
+      validateNestedIdsExist(updatedField._2);
+    }
+
+    for (StructField addedField : schemaChanges.addedFields()) {
+      validateNestedIdsExist(addedField);
+    }
+
+    validateNestedIdConsistency(currentFieldsById, updatedFieldsById);
+  }
+
+  private static void validateNestedIdsExist(StructField field) {
+    if (field.getDataType() instanceof MapType || field.getDataType() instanceof ArrayType) {
+      checkArgument(
+          hasNestedColumnIds(field),
+          "Field %s is missing expected nested column IDs",
+          field.getName());
+      if (field.getDataType() instanceof MapType) {
+        checkArgument(
+            getNestedFieldIds(field).size() >= 2,
+            "Map field %s needs at least 2 nested IDs, one for the key and one for the value",
+            field.getName());
+      }
+    }
+  }
+
+  /* Mapping of field ID to nested IDs for that field */
+  private static Map<Long, List<NestedIdEntry>> columnIdToNestedIds(
```

Comment:
nit: probably just a noise comment? if anything -- you could tweak the function name to be clearer -- but I don't think this comment adds anything?

Aside: I've recently been reading Clean Code by Robert C Martin -- I'm trying to apply some of those learnings :p

Reply:
I think that's a great suggestion, I feel the same way after reading it again tbh, the comment isn't adding anything. I'll remove it